core/shared/utils/bh_log.c

```diff
@@ -11,6 +11,18 @@
  */
 static uint32 log_verbose_level = BH_LOG_LEVEL_WARNING;
 
+#ifndef BH_PLATFORM_LINUX_SGX
+int (*ptr_vprintf)(const char *, va_list) = vprintf;
```

Comment:
This will make the platform layer depends on the utils, which is not reasonable.

Had better put the `ptr_vprintf` to every platform.

Also the `bh_set_vprintf` should be moved to every platform's directory and rename to `os_set_vprintf`

Of course this will introduce some duplicate code, but avoid breaking the architecture

Reply:
Yes, had better declare `os_set_vprintf` in `platform_api_vmcore.h`, and implement it in platform layer besides with `os_printf` and `os_vprintf`.